### PR TITLE
Auto-format all files

### PR DESCRIPTION
**Pull request by aabounegm**
This PR runs the new auto-formatter for Rzk that implements the [style guide](https://github.com/rzk-lang/sHoTT/blob/main/src/STYLEGUIDE.md). It is currently still under work in https://github.com/rzk-lang/rzk/pull/144, so before it is merged and released, I'd like to make sure that it works as expected, so I ran it on all files in this project (using `rzk format --write`). This should help find implementation mistakes in the formatter and/or inconsistencies in the style guide (the hopeful best-case scenario would be that it aligns all formalizations with the current style guide 🙂).

Please review these changes and point out any problems or possible improvements.

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
Awesome! I will take the time to review this later this week.

File: src/hott/02-homotopies.rzk.md
Comment (fredrik-bakke) [inline src/hott/02-homotopies.rzk.md:193]: The arrow could be two spaces less indented here

File: src/hott/02-homotopies.rzk.md
Comment (fredrik-bakke) [inline src/hott/02-homotopies.rzk.md:167]: This space should not be automatically removed since it is only preceded by single-character tokens.

File: src/hott/06-contractible.rzk.md
Comment (fredrik-bakke) [inline src/hott/06-contractible.rzk.md:94]: The convention here should be to align the comma with the outer parenthesis

File: src/hott/06-contractible.rzk.md
Comment (fredrik-bakke) [inline src/hott/06-contractible.rzk.md:241]: The space between `(` and `\` should be kept for the same reason as above.

File: src/hott/02-homotopies.rzk.md
Comment (fredrik-bakke) [reply src/hott/02-homotopies.rzk.md:193]: I may also note that I didn't intend for the convention to be that the lambda arrow should be placed on the next line as done here. That's not to say I'm against that convention.

Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md:193]: Right, one limitation with the current implementation is that it does not check whether a given line is indented properly. We will hopefully find an easy way to fix indentations in general.
About the lambda arrow, I seem to have missed that point in the guide and will fix it now.

File: src/hott/02-homotopies.rzk.md
Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md:167]: Perhaps the style guide should clarify this point further as it is not easy to check for the "opening parentheses marking a branch" with just lexical analysis. I was confused about this point as I saw examples of both styles, such as:
```rzk
Σ ( is-segal-A : is-segal A)
  , ( (x : A)
```
Note that all tokens before `x` are single-character, yet there is no space before it. Same with `( (f , g) : iff A B)`, for example.

Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md:167]: While working on this, I also noticed that the rule as you described it could also lead to such output:
```rzk
  ( C : ( x : A) → (y : B) → U)
  ( f : ( x : A) → (y : B) → C x y)
```
But I don't think it is desirable to add the space before the `x`, right?
I think this rule needs a more precise specification to reliably implement it in code

Comment (fredrik-bakke) [reply src/hott/02-homotopies.rzk.md:167]: Apologies, my description was not entirely accurate. I agree with you.

Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md:167]: No worries, just let me know once a stricter definition is agreed upon.
In the meantime, if all else is good, should I just disable this rule (temporarily) in the formatter, or did you find other problems? (Or should I wait to implement this rule correctly before merging?)

File: src/hott/02-homotopies.rzk.md
Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md]: As a temporary fix until proper indentation is implemented, binary operators now move to an indentation level that is one less than the next line originally was.

File: src/hott/02-homotopies.rzk.md
Comment (fredrik-bakke) [reply src/hott/02-homotopies.rzk.md]: Sorry, I didn't see this question until now. Let me think quickly

Comment (aabounegm) [reply src/hott/02-homotopies.rzk.md]: Fixed by checking if the previous tokens consist solely of the following: `["(", ":", ",", "=", "→", "↦", "≡", "≤", "∧", "∨", "Σ", "Σ", "*₁", "0₂", "1₂", "×"]` (almost any built-in single-character token)

Comment (fredrik-bakke) [reply src/hott/02-homotopies.rzk.md]: Oh, that seems reasonable

File: src/hott/01-paths.rzk.md
Comment (fredrik-bakke) [inline src/hott/01-paths.rzk.md:205]: Here I don't think spaces should be inserted, but doing so is fine as the parentheses shouldn't actually be there in my opinion.

Comment (fredrik-bakke) [reply src/hott/01-paths.rzk.md:205]: Ah, cool! If you don't mind, maybe I can share one specific opinion on the matter. Namely, I'd be careful to make the linter too obnoxious about "redundant" parentheses. At times, parentheses may be redundant in the sense that the code parses the same way to the typechecker without them, but they may still convey useful information to the reader of the code. For instance, if you have defined concatenation of identifications to be left associative, then you could formulate the associator to have type
```text
  p ∙ q ∙ r = p ∙ (q ∙ r),
```
but this looks rather ridiculous. Of course, in this hypothetical scenario, you want to properly parenthesize this expression as
```text
  (p ∙ q) ∙ r = p ∙ (q ∙ r).
``` 
Personally, I would find it quite annoying if the linter told me that this parenthesis was redundant, because it isn't.

File: src/hott/03-equivalences.rzk.md
Comment (fredrik-bakke) [inline src/hott/03-equivalences.rzk.md:278]: This one's misplaced. It should be one level more indented.

Comment (fredrik-bakke) [reply src/hott/03-equivalences.rzk.md:278]: The last line, that is

File: src/hott/01-paths.rzk.md
Comment (fredrik-bakke) [inline src/hott/01-paths.rzk.md:312]: Hmm, this looks a little strange to me with how the placement of `→` is inconsistent between lambdas and Pis, but this is indeed the convention. It's up to the larger community if they want to change the convention though.

File: src/hott/06-contractible.rzk.md
Comment (fredrik-bakke) [inline src/hott/06-contractible.rzk.md:245]: This should be
```suggestion
      , \ b →
        all-elements-equal-is-contr B is-contr-B
          ( f (center-contraction A is-contr-A)) b))
```

File: src/hott/06-contractible.rzk.md
Comment (fredrik-bakke) [inline src/hott/06-contractible.rzk.md:339]: this is also wrong and should be formatted similarly to my previous comment.

File: src/hott/06-contractible.rzk.md
Comment (fredrik-bakke) [inline src/hott/06-contractible.rzk.md:477]: This should be
```suggestion
        ( ( \ b x →
            ( transport A B
              ( center-contraction A is-contr-A) x
              ( realign-homotopy-contraction A is-contr-A x) b))
```
(disregarding the redundant parentheses)

File: src/hott/04-half-adjoint-equivalences.rzk.md
Comment (fredrik-bakke) [inline src/hott/04-half-adjoint-equivalences.rzk.md:779]: Oh, I forgot to mention, the convention is that after the walrus separator, code starts with the same indentation level:
```suggestion
  :=
  ( has-retraction-ap-is-half-adjoint-equiv x y
  , has-section-ap-is-half-adjoint-equiv x y)
```

File: src/hott/04-half-adjoint-equivalences.rzk.md
Comment (fredrik-bakke) [inline src/hott/04-half-adjoint-equivalences.rzk.md:331]: This looks like a good test case because it incorporates a lot of different indentation levels:
1. indentation starts on the same level as `:=`
2. lambda indentation
3. on the (second to) top level, arguments have the same indentation level as the function they are applied to

```suggestion
  :=
  \ b →
  concat B
  ( ( section-composite-has-inverse A B f has-inverse-f) b)
  ( ( section-composite-has-inverse A B f has-inverse-f)
    ( ( section-composite-has-inverse A B f has-inverse-f) b))
  ( b)
  ( rev B
    ( ( section-composite-has-inverse A B f has-inverse-f)
      ( ( section-composite-has-inverse A B f has-inverse-f) b))
    ( ( section-composite-has-inverse A B f has-inverse-f) b)
    ( has-inverse-discarded-htpy A B f has-inverse-f
      ( ( section-composite-has-inverse A B f has-inverse-f) b)))
  ( concat B
    ( ( section-composite-has-inverse A B f has-inverse-f)
      ( ( section-composite-has-inverse A B f has-inverse-f) b))
    ( ( section-composite-has-inverse A B f has-inverse-f) b)
    ( b)
    ( ap A B
      ( ( retraction-composite-has-inverse A B f has-inverse-f)
        ( map-inverse-has-inverse A B f has-inverse-f b))
      ( map-inverse-has-inverse A B f has-inverse-f b) f
      ( ( first (second has-inverse-f))
        ( map-inverse-has-inverse A B f has-inverse-f b)))
    ( ( has-inverse-discarded-htpy A B f has-inverse-f b)))
```

File: src/simplicial-hott/03-extension-types.rzk.md
Comment (fredrik-bakke) [inline src/simplicial-hott/03-extension-types.rzk.md:134]: These spaces also look unnecessary and weird to me, but again it's a matter of preference.

File: src/simplicial-hott/03-extension-types.rzk.md
Comment (fredrik-bakke) [inline src/simplicial-hott/03-extension-types.rzk.md:1118]: Since the definition is more than one line long, a newline should be inserted directly after `:=` (also note that the indentation is wrong in this example)

File: src/simplicial-hott/12-cocartesian.rzk.md
Comment (fredrik-bakke) [inline src/simplicial-hott/12-cocartesian.rzk.md:74]: Presumably you would want to align it like this:
```suggestion
      ( b'' : B) → (v : hom B b' b'') → (w : hom B b b'')
    → ( sigma : hom2 B b b' b'' u v w) → (e'' : P b'')
    → ( h : dhom B b b'' w P e e'')
    → is-contr
        ( Σ ( g : dhom B b' b'' v P e' e'')
        , ( dhom2 B b b' b'' u v w sigma P e e' e'' f g h))
```
Although I don't know if that should be a rule. I wasn't the one to initially introduce the prepending binary operators convention.

**Review by fredrik-bakke — COMMENTED** [review]
Okay, I've reviewed about half the files now and pointed out different issues I saw. Although the autoformatter isn't perfect yet, I think it is a great improvement over no autoformatter, and so would be happy to have it deployed even in its current revision. Thanks for taking the initiative on this! And let me know if/when you want further input :)

File: src/hott/01-paths.rzk.md
Comment (aabounegm) [reply src/hott/01-paths.rzk.md:205]: That's a good point and we should definitely consider. Once the linter is added to the backlog, we will surely ask for feedback on what rules it should or shouldn't have

File: src/hott/06-contractible.rzk.md
Comment (aabounegm) [reply src/hott/06-contractible.rzk.md:339]: Does that mean that there should **always** be a new line after the lambda arrow?

Comment (fredrik-bakke) [reply src/hott/06-contractible.rzk.md:339]: No, only if the full lambda expression does not fit within a single line.

Comment (fredrik-bakke) [reply src/hott/06-contractible.rzk.md:339]: E.g. `\ x → x` is totally fine, but `\ x → impossibly-long-expression-that-surpasses-80-character-limit` is not.

File: src/simplicial-hott/12-cocartesian.rzk.md
Comment (aabounegm) [reply src/simplicial-hott/12-cocartesian.rzk.md:74]: And that is actually a good example of why indentation should be left up to the user and not handled automatically

Comment (aabounegm) [reply src/simplicial-hott/12-cocartesian.rzk.md:74]: Ah, that's a good catch. The formatter does not know that the comma belongs to the `Σ` and not the `(` and applies the rule blindly. If it's fine by you, we can consider it an edge case for now and part of the "smart indentation" which is not implemented yet :)

File: src/hott/07-fibers.rzk.md
Comment (aabounegm) [reply src/hott/07-fibers.rzk.md:79]: But why? There aren't any multi-character tokens before the second parenthesis

File: src/hott/06-contractible.rzk.md
Comment (aabounegm) [reply src/hott/06-contractible.rzk.md:477]: Sadly, the indentation has to be adjusted manually for now. There is still some debate over whether the formatter should even enforce indentation or leave it up to the user.

File: src/simplicial-hott/12-cocartesian.rzk.md
Comment (fredrik-bakke) [reply src/simplicial-hott/12-cocartesian.rzk.md:74]: Fine by me

File: src/hott/07-fibers.rzk.md
Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: Well...

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: The point of the space is to help highlight the tree structure of definitions. It is not doing that here. Somewhere along the way, this point lost its significance, and conventions like adding spaces in the type signatures for code alignment were added. I consider that a different convention, although they are easy to conflate.

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: If the code spanned multiple lines, i.e.
```text
  ( ( my-incredibly-long-variable-name ,
      my-other-incredibly-long-variable-name) :
    rev-fib A B f b)
```
_then_ the space should be there because it helps highlight the tree structure.

Comment (aabounegm) [reply src/hott/07-fibers.rzk.md:79]: But isn't it theoretically possible to perform deeply nested pattern matching, which one might want to wrap to the next lines?

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: Oh, but with the current conventions this should be:
```suggestion
  ( ( my-incredibly-long-variable-name
    , my-other-incredibly-long-variable-name)
    : rev-fib A B f b)
```

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: > But isn't it theoretically possible to perform deeply nested pattern matching, which one might want to wrap to the next lines?
> 
> **Edit**: Ah, your other comment just loaded now 😅. The thing is that it is not easy to determine beforehand whether the code will span multiple lines, so the space is added anyway for consistency

Although I can sympathize with your situation, I think it is unfortunate to introduce conventions based on the implementation details of the autoformatter.

Comment (aabounegm) [reply src/hott/07-fibers.rzk.md:79]: I agree with you for sure, but then does that also apply to the "tree structure" rule everywhere? In other words, should there never be a space after an opening parenthesis unless it is part of a multi-line expression with some nested expressions?

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: It also applies at the leaves

Comment (fredrik-bakke) [reply src/hott/07-fibers.rzk.md:79]: But the conventions regarding where to place such spaces have taken on a life of their own, and I don't know what others in the community want.

File: src/hott/03-equivalences.rzk.md
Comment (fizruk) [reply src/hott/03-equivalences.rzk.md:278]: This is due to incorrect indentation in the original code. The formatter does not fix that (at least not at the moment).

**Comment by aabounegm** [comment]
@fredrik-bakke Thank you very much for the constructive feedback! I hope that most of the issues have been addressed now. It seems to me that the remaining unresolved conversations are about either inserting line breaks or fixing indentation, both of which are planned for the future as they are not straightforward to implement with the current architecture (opened issues https://github.com/rzk-lang/rzk/issues/147 and https://github.com/rzk-lang/rzk/issues/148).
I do not have any other changes planned for this PR, so if you think it is ready to be merged then please approve it so I can merge 😁

**Review by fredrik-bakke — APPROVED** [review]
Excellent! Looks good to me :) Thanks again for doing this.

**Pull request merged**
